src/fmu/tools/fipmapper/fipmapper.py

```diff
@@ -0,0 +1,320 @@
+ * A zone is assumed to be present for all regions, but not relevant in this class
+
+"""
+from pathlib import Path
+import logging
+from typing import Union, Dict, List, Any, Optional
+
+import yaml
+
+logger = logging.getLogger(__name__)
+
+
+class FipMapper:
+    def __init__(
+        self,
+        yamlfile: Union[str, Path] = None,
+        mapdata: Dict[str, str] = None,
+        skipstring: Union[List[str], str] = None,
+    ):
+        """FipMapper is a utility class for being able to map between
+        regions/zones in the geomodel (RMS) and to different region divisions in the
+        dynamic model (Eclipse).
+
+        Primary usage is to determine which RMS regions corresponds to
+        which FIPNUMs, similarly for zones, and in both directions
+
+        Configuration is via a yaml-file or directly with a dictionary.
+
+        Several data structures in the dictionary can be used, such that
+        the needed information can be extracted from the global configurations
+        file.
+
+        Args:
+            yamlfile: Filename
+            mapdata: direct dictionary input. Provide only one of the
+                arguments, not both.
+            skipstring: List of strings which will be ignored (e.g. ["Totals"]).
+        """
+        self._mapdata: Dict[str, dict] = {}  # To be filled with data we need.
+
+        if skipstring is None:
+            self.skipstring = []
+        if isinstance(skipstring, str):
+            self.skipstring = [skipstring]
+
+        if yamlfile is not None and mapdata is not None:
+            raise ValueError(
+                "Initialize with either yamlfile or explicit data, not both"
+            )
+        if yamlfile is None and mapdata is None:
+            logger.warning("FipMapper initialized with no data")
+
+        if yamlfile is not None:
+            logger.info("Loading data from %s", yamlfile)
+            with open(yamlfile) as stream:
+                yamldata = yaml.safe_load(stream)
+            logger.debug(str(yamldata))
+        else:
+            yamldata = mapdata
+
+        if yamldata is not None:
+            self._get_explicit_mapdata(yamldata)
+
+        if yamldata is not None and "global" in yamldata:
+            # This is a fmu-config file.
+            self._fipdata_from_fmuconfigyaml(yamldata)
+
+        # Webviz YML format:
+        if yamldata is not None and "FIPNUM" in yamldata:
+            self._fipdata_from_webvizyaml(yamldata)
+
+        assert isinstance(self._mapdata, dict), "FipMapper needs a dictionary"
+
+        # Determine our capabilities:
+        self.has_fip2region = "fipnum2region" in self._mapdata
+        self.has_fip2zone = "fipnum2zone" in self._mapdata
+        self.has_region2fip = "region2fipnum" in self._mapdata
+        self.has_zone2fip = "zone2fipnum" in self._mapdata
+
+    def _get_explicit_mapdata(self, yamldata: Dict[str, Any]):
+        """Fetch explicit mapping configuration from a dictionary,
```

Comment:
End first sentence with period ., not comma ,

Reply:
Fixed.